firedrake/constant.py

```diff
@@ -21,8 +21,10 @@ def _globalify(value):
         dat = op2.Global(shape, data)
     elif rank == 2:
         dat = op2.Global(shape, data)
+    elif rank == 3:
+        dat = op2.Global(shape, data)
     else:
-        raise RuntimeError("Don't know how to make Constant from data with rank %d" % rank)
+        raise RuntimeError("Don't know how to make Constant from data with Tensor rank (order) %d" % rank)
```

Comment:
Thanks, I think we can probably support arbitrary rank, so perhaps we can simplify all of this to:

```
if rank = len(shape):
    dat = op2.Global(1, data)
else:
   dat = op2.Global(shape, data)
```